### PR TITLE
Intro tour for classic admin interface

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/dotcom-forge/issues/7404

## Proposed changes:

Adds a dismissible intro tour which displays some educational tooltips for folks who turn the classic admin interface on:

Step 1 | Step 2 | Step 3
--- | --- | ---
Screenshot 2024-05-27 at 14 45 19 | Screenshot 2024-05-27 at 14 45 25 | Screenshot 2024-05-27 at 14 45 32

The tour shows up in the WP Admin dashboard when the `?admin-interface-changed=true` query param is present, which (as of https://github.com/Automattic/wp-calypso/pull/91102) is automatically added to the URL after activating the classic admin interface.

We also display a success notification to indicate that the admin interface has changed:

Screenshot 2024-05-27 at 14 47 30

**Some considerations** 
- The intro tour is hidden on mobile viewports
- The tooltips always have the same dark color regardless of the color scheme selected on the site
- The tooltips are not repositioned if the sidebar or the site management panel card are collapsed. The effort needed to address this is probably not worth the number of users who would face this issue
- The copy of the tour is being discussed here: pfsHM7-17R-p2

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply these changes to your WP.com testing site
- Go to wordpress.com/settings/general and select your site
- Change the admin interface to Classic
- You should land in `/wp-admin/?admin-interface-changed=true`
- Make sure the intro tour shows up
- Make sure you can proceed to next steps
- Make sure they are placed correctly
- Make sure you the tour is dismissed after clicking on the last step's CTA (or after clicking on the X button that appears in every step)
- Make sure the tour no longer shows up after being dismissed (the success notification should still be displayed)

If you need the tour to be visible after dismissing it, delete the `wpcom_classic_tour_completed` option.

### CODE REVIEW

**Comment by rcrdortiz** [comment]
If I navigate away from `/wp-admin?admin-interface-changed=true` the tour isn't displayed/completed. Maybe we can add an option or cookie to check if the tour is in progress and continue to show it across screens?

**Comment by mmtr** [comment]
> If I navigate away from `/wp-admin?admin-interface-changed=true` the tour isn't displayed/completed. Maybe we can add an option or cookie to check if the tour is in progress and continue to show it across screens?

It's not that easy, because there is one step of the tour that requires to be in the dashboard (step 2: Hosting overview).

**Comment by rcrdortiz** [comment]
When the notice is displayed, and I move to step 2 and then reload, the tour starts at step 1 again.

**Comment by rcrdortiz** [comment]
> > If I navigate away from `/wp-admin?admin-interface-changed=true` the tour isn't displayed/completed. Maybe we can add an option or cookie to check if the tour is in progress and continue to show it across screens?
> 
> It's not that easy, because there is one step of the tour that requires to be in the dashboard (step 2: Hosting overview).

Maybe we can skip that step if we're not on `/wp-admin`? I mean, if the user navigates away he's never going to see it anyways right?

**Comment by rcrdortiz** [comment]
[Minor]
When we resize the screen the current tour step doesn't adjust to the new screen size until re-rendered.

https://github.com/Automattic/jetpack/assets/1989914/a0abc141-494f-47d6-893b-f0d05b18199e

**Comment by mmtr** [comment]
> When the notice is displayed, and I move to step 2 and then reload, the tour starts at step 1 again.

Yup, that's expected for now. It's considered an uncommon case that is not worth handling.

> Maybe we can skip that step if we're not on /wp-admin? I mean, if the user navigates away he's never going to see it anyways right?

If we do that, and they complete the tour outside of the dashboard, they'll never see the step 2 even when they are in the dashboard. 

With the current approach, they won't see the tour outside of the dashboard, but they will be able to see the step 2 when they are in the dashboard.

Anyway, the effort needed to maintain the tour across screens is likely no worth it considering that (1) navigating away when the tour is present probably won't be a common case, (2) the tour is just educational – it's ok if it's not followed, and (3) the tour will still be there for those who navigate away and go back to the dashboard.

**Comment by rcrdortiz** [comment]
> (3) the tour will still be there for those who navigate away and go back to the dashboard.

This statement is misleading, by `go back` you mean navigating using the back button and not actually navigating to the dashboard again right? If I click on `Appearance` and then navigate back to the dashboard by clicking on the `Dashboard` menu, the tour isn't shown.

**Comment by mmtr** [comment]
> When we resize the screen the current tour step doesn't adjust to the new screen size until re-rendered.

Yeah, that's actually a WP bug with the resizing. If you reload the page with that screen size, you'll see that the sidebar is visible and does't not transition between collapsed/expanded status:

https://github.com/Automattic/jetpack/assets/1233880/6c3df83b-96ac-40f5-ab97-ef40e660519a

Considering that users usually don't resize browsers (at least not in a significant way that changes the viewport from desktop to mobile or viceversa), I thought it wasn't worth to handle this issue.

**Comment by mmtr** [comment]
> If I click on Appearance and then navigate back to the dashboard by clicking on the Dashboard menu, the tour isn't shown.

That's good catch! I'll push a change to keep showing the tour in that case.

**Review by rcrdortiz — DISMISSED** [review]
Code LGTM, on the functional side there are a few things that could be improved but I won't block this PR since the current implementation adds value.

**Comment by mmtr** [comment]
> If I click on Appearance and then navigate back to the dashboard by clicking on the Dashboard menu, the tour isn't shown.

Solved in https://github.com/Automattic/jetpack/pull/37533/commits/67d8181c5611272ec090542f8feb7767b6bda06e

**Review by rcrdortiz — APPROVED** [review]
Thanks for persisting the tour's state when navigating!

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php [hunk 1]
```diff
@@ -376,137 +376,6 @@ function wpcom_add_jetpack_menu_item() {
 }
 add_action( 'jetpack_admin_menu', 'wpcom_add_jetpack_menu_item' );
 
-/**
- * Helper function to determine if the admin notice should be shown.
- *
- * @return bool
- */
-function wpcom_site_menu_should_show_notice() {
```
Comment (arthur791004) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php:384]: Are these changes are intentional 🤔

Comment (mmtr) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php:384]: Yes, this notice was displaying the same info as the first step of the new guided tour

**Pull request merged**